ArmRobot/Assets/Scripts/RandomizerPositionObject.cs

```diff
@@ -0,0 +1,166 @@
+
+    Bounds tableBounds;
+    public float yAltitudeTable = 0.778f;
+
+
+    // CONTROL
+
+    public void Move(List<GameObject> listOfAlreadyMovedObjects)
+    {
+        // random position (on table, within reach)     
+        Vector2 tableTopPoint = RandomReachablePointOnTable(listOfAlreadyMovedObjects);
+        Vector3 tableCenter = tableBounds.center;
+        float x = tableCenter.x + tableTopPoint.x;
+        float z = tableCenter.z + tableTopPoint.y;
+        transform.position = new Vector3(x, yAltitudeTable, z);
+        
+
+        // random rotation
+        Vector3 randomRotation = new Vector3(
+            transform.rotation.eulerAngles.x,
+            Random.value * 360.0f,
+            transform.rotation.eulerAngles.z);
+        transform.rotation = Quaternion.Euler(randomRotation);
+    }
+
+
+    // HELPERS
+
+    Vector2 RandomReachablePointOnTable(List<GameObject> listOfAlreadyMovedObjects)
+    {
+        while (true)
+        {
+            Vector2 randomOffset = RandomPoint(robotMinReach, robotMaxReach, listOfAlreadyMovedObjects);
+            bool onTable = PointOnTable(randomOffset);
+            if (onTable)
+            {
+                return randomOffset;
+            }
+        }
+    }
+
+    bool PointOnTable(Vector2 point)
+    {
+        /*  point: The 2D point on table top, relative to center of table top.
+         *  Determines if this point would be on the table or not.      
+         */
+        GameObject table = GameObject.Find("Table");
+        tableBounds = table.GetComponent<Collider>().bounds;
+        Vector3 tableExtents = tableBounds.extents;
+        float targetRadius = 0f;
+        if (tag == "Sphere"){
+                targetRadius = GameObjectRadius(gameObject); // for the sphere 
+            } 
+        else if (tag == "Cylinder"){
+                targetRadius = GameObjectRadius(gameObject); 
+            }
+
+        float safeXDistance = tableExtents.x - targetRadius;
+        float safeZDistance = tableExtents.z - targetRadius;
+        float xDistance = Mathf.Abs(point.x);
+        float yDistance = Mathf.Abs(point.y);
+    
+        bool onTable = (xDistance < safeXDistance) && (yDistance < safeZDistance);
+        return onTable;
+    }
+
+    public Vector2 RandomPoint(float minRadius, float maxRadius, List<GameObject> listOfAlreadyMovedObjects)
+    {
+        /*  Picks a 2D point randomly at uniform. Must be between minRadius and
+         *  maxRadius from center and not at the same place than the cube. 
+         *  Point given relative to center.      
+         */
+
+        while (true)
+        {
+            // pick a random point between the square of edge (2 * minRadius + radiusOfTheGameObject) and the one with 
+            // edge (2 * maxRadius - radiusOfTheGameObject) to be sure that the point is reachable by the robot 
+            float gameObjectRadius = GameObjectRadius(gameObject);
+            float randomX = 0.25f;
+            float randomZ = 0.25f;
```

Comment:
Why do you set `randomX` and `randomZ` to 0.25 here when you just reset them a few lines below? Seems like you could remove these lines.